src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -114,6 +125,22 @@
                       "/etc/default/puppetserver on Debian systems.")))))))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Helper Functions
+
+(defn construct-404-error-message
+  [jruby-service]
+  (str "Error: Invalid URL - Puppet Server expects requests that conform to the "
+       "/puppet and /puppet-ca APIs.\n\n"
+       "Note that Puppet 3 agents aren't compatible with this version; if you're "
+       "running Puppet 3, you must either upgrade your agents to match the server "
+       "or point them to a server running Puppet 3.\n\n"
+       "Server Info:\n"
+       "  Puppet Server version: " (version-check/get-version-string "puppet-server") "\n"
+       "  Puppet version: " (:puppet-version (config/get-puppet-config jruby-service)) "\n"
```

Comment:
@cprice404 I've created (SERVER-334)[https://tickets.puppetlabs.com/browse/SERVER-334] to track this. It'll be a really easy fix so I went ahead and pulled it into the current sprint and will knock it out today.